osu.Game/Screens/Edit/Compose/Components/SelectionBoxRotationHandle.cs

```diff
@@ -33,10 +48,59 @@ private void load()
             });
         }
 
+        protected override void LoadComplete()
+        {
+            base.LoadComplete();
+            cumulativeRotation.BindValueChanged(_ => updateTooltipText(), true);
+        }
+
         protected override void UpdateHoverState()
         {
             base.UpdateHoverState();
             icon.FadeColour(!IsHeld && IsHovered ? Color4.White : Color4.Black, TRANSFORM_DURATION, Easing.OutQuint);
         }
+
+        protected override bool OnDragStart(DragStartEvent e)
+        {
+            bool handle = base.OnDragStart(e);
+            if (handle)
+                cumulativeRotation.Value = 0;
+            return handle;
+        }
+
+        protected override void OnDrag(DragEvent e)
+        {
+            base.OnDrag(e);
+
+            float instantaneousAngle = convertDragEventToAngleOfRotation(e);
+            cumulativeRotation.Value += instantaneousAngle;
+
+            if (cumulativeRotation.Value < -180)
+                cumulativeRotation.Value += 360;
+            else if (cumulativeRotation.Value > 180)
+                cumulativeRotation.Value -= 360;
+
+            HandleRotate?.Invoke(instantaneousAngle);
+        }
+
+        protected override void OnDragEnd(DragEndEvent e)
+        {
+            base.OnDragEnd(e);
+            cumulativeRotation.Value = null;
+        }
+
+        private float convertDragEventToAngleOfRotation(DragEvent e)
+        {
+            // Adjust coordinate system to the center of SelectionBox
+            float startAngle = MathF.Atan2(e.LastMousePosition.Y - selectionBox.DrawHeight / 2, e.LastMousePosition.X - selectionBox.DrawWidth / 2);
+            float endAngle = MathF.Atan2(e.MousePosition.Y - selectionBox.DrawHeight / 2, e.MousePosition.X - selectionBox.DrawWidth / 2);
+
+            return (endAngle - startAngle) * 180 / MathF.PI;
+        }
+
+        private void updateTooltipText()
+        {
+            TooltipText = cumulativeRotation.Value?.ToLocalisableString("0.0°") ?? default(LocalisableString);
```

Comment:
Curious if `default(LocalisableString)` works any differently from leaving it null. In testing I couldn't seem to make it produce different results.

Reply:
It will work the same, but using `default` prevents an implicit conversion. Not a huge deal either way.